src/h_vialib/secure/expiry.py

```diff
@@ -0,0 +1,81 @@
+"""Functions for calculating expiry."""
+
+from datetime import datetime, timedelta, timezone
+
+# An arbitrary time in the past to quantize our expiry times to
+YEAR_ZERO = datetime(year=2020, month=1, day=1, tzinfo=timezone.utc)
+
+
+def quantized_expiry(max_age, divisions=2):
+    """Create a quantized expiry time.
+
+    The time returned is guaranteed be:
+        * A multiple of `max_age / divisions`
+        * At most `max_age` seconds in the future
+        * At least `max_age * (divisions - 1) / divisions` seconds in the future
+
+    :param max_age: The maximum age to issue (int, or timedelta)
+    :param divisions: How many subdivisions of the max age for quantization
+    :return: A datetime object
```

Comment:
I think the key thing about this is that if you call `quantized_expiry()` repeatedly with the same arguments it will keep on returning the same value until some amount of time (depending on `max_age` and `divisions`) has passed, then it'll start returning a different value, right? Rather than returning a different value every time you call it a millisecond later.

The seems like the key guarantee here / the reason this function exists so worth explaining in the docstring

Reply:
I've added some more of the why, rather than just the what

Reply:
Should also add this to `h_vialib/__init__.py` since at least Via 3 is gonna be using it (just to make it more convenient to import)